app/src/main/java/com/vestrel00/daggerbutterknifemvp/ui/common/BaseFragment.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2017 Vandolf Estrellado
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.vestrel00.daggerbutterknifemvp.ui.common;
+
+import android.app.Fragment;
+import android.app.FragmentManager;
+import android.content.Context;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import dagger.android.AndroidInjection;
+import dagger.android.AndroidInjector;
+import dagger.android.DispatchingAndroidInjector;
+import dagger.android.HasFragmentInjector;
+
+/**
+ * Abstract Fragment for all Fragments and child Fragments to extend. This contains some boilerplate
+ * dependency injection code and activity {@link Context}.
+ * <p>
+ * <b>DEPENDENCY INJECTION</b>
+ * We could extend {@link dagger.android.DaggerFragment} so we can get the boilerplate
+ * dagger code for free. However, we want to avoid inheritance (if possible and it is in this case)
+ * so that we have to option to inherit from something else later on if needed.
+ * <p>
+ * <b>VIEW BINDING</b>
+ * This fragment handles view bind and unbinding.
+ */
+public abstract class BaseFragment extends Fragment implements HasFragmentInjector {
```

Comment:
The latest branches of this repo have `BaseFragment` subclassing `DialogFragment` instead of `Fragment` in order to flawlessly support `DialogFragment` injection. See this [PR](https://github.com/vestrel00/android-dagger-butterknife-mvp/pull/64) for more details.